dynosaur_derive/src/lib.rs

```diff
@@ -333,6 +340,26 @@ fn mk_dyn_struct_impl_item(struct_ident: &Ident, item_trait: &ItemTrait) -> Toke
                                 fut
                             }
                         }
+                    } else if is_rpit(&sig) {
+                        let ret = expand_ret_ty(&sig);
+
+                        if has_lifetime_bound(&sig) {
+                            quote! {
+                                #sig {
+                                    let ret: Box<dyn #ret> = self.ptr.#ident(#(#args),*);
+                                    let ret: Box<dyn #ret> = unsafe { ::core::mem::transmute(ret) };
+                                    ret
```

Comment:
This doesn't make sense, is it exercised anywhere? I would expect it to be exercised by non-future-impl-traits.rs above, but the code there doesn't match this code, so not sure what's going on.

In any case I think we want `'_` in both cases; in other words, there's no need for this if/else to exist.

Reply:
Yes it is. The Iterator example would be affected.
We have `Iterator<Item=u8> + '_`. `has_lifetime_bound` returns true and in this case, we just do the `#ret` version. `#ret` already includes '_ as bound so it would give you back `Iterator<Item=u8> + '_` which is what we want.

If we were going with the `#ret + '_` we would end with `Iterator<Item=u8> + '_ + '_` which is wrong.
This is actually the point and why `has_lifetime_bound` exists.

Reply:
Ah, I suspect this won't work if the bound is anything other than `'_` then. If it were `'a` you would end up with an error.

Replacing the bound with `'_` should always work though.

Reply:
Fixed